### PR TITLE
first and third party script info

### PR DESCRIPTION
**Pull request by jhadev**
Example output

CleanShot 2022-05-26 at 20 27 51@2x

### CODE REVIEW

**Comment by nucliweb** [comment]
We need to improve the host validation.

A lot of sites use a subdomain for the resources (as a custom CDN), and these resources doesn't Thrid-party.

e.g. On this site, all the resources use a subdomain, and the script detects all the First Party Scripts as Third Party Scripts.

image

**Comment by jhadev** [comment]
> 

You are right. Technically they aren't third party despite the fact they usually incur network cost. So the wording is kinda deceiving in this case. But you are totally right and like your idea.

**Comment by jhadev** [comment]
Example using katespade.com just bc I knew a subdomain.
Image

File: README.md [hunk 1]
```diff
@@ -175,4 +175,72 @@ function getImgs(sortBy) {
...
+  const third = [];
+
+  resourceListEntries.forEach((resource) => {
+    // check for initiator type
+    const value = "initiatorType" in resource;
+    if (value) {
+      if (resource.initiatorType === "script") {
+        const { host } = new URL(resource.name);
+        // check if resource url host matches location.host = first party script
+        if (host === location.host || HOSTS.includes(host)) {
```
Comment (nucliweb) [inline README.md:206]: 👏

File: README.md [hunk 2]
```diff
@@ -175,4 +175,72 @@ function getImgs(sortBy) {
...
+```js
+// ex: katespade.com - list firsty party subdomains in HOSTS array
+const HOSTS = ["assets.katespade.com"];
+
+function getScriptInfo() {
+  const resourceListEntries = performance.getEntriesByType("resource");
+  // set for first party scripts
+  const first = [];
+  // set for third party scripts
+  const third = [];
```
Comment (nucliweb) [inline README.md:197]: ❤️

**Review by nucliweb — APPROVED** [review]

**Pull request merged**